sycl/doc/SYCLPluginInterface.md

```diff
@@ -0,0 +1,178 @@
+in the ICD loader and records the returned platform objects for later use. The
+plugin may choose to dynamically link with libOpenCL.so at build time.
+ 
+##### Linking to ICD-incompatible OpenCL installations
+
+For the OpenCL plugin to hook up such installation, it must be described in a
+configuration file similarly to plugin configuration file. *TBD - location and
+format of custom OpenCL installation config file*. The plugin will dynamically
+load the dynamic library pointed to in the configuration file and query it for
+OpenCL API function addresses similarly ICD loader. In more details, it will:
+- search for  the ```clGetExtensionFunctionAddress``` function in the 
+  library and call it with the string *“clIcdGetPlatformIDsKHR”* to find
+  the ```clIcdGetPlatformIDsKHR``` function pointer
+- then, using ```clIcdGetPlatformIDsKHR```, find the set of
+  ```cl_platform_id``` objects that the library provides, and store them
+  internally to enumerate available devices when required
+
+### Device enumeration by plugins
+
+TBD
+
+## PI API Specification
+
+PI allows for two levels of API implementation - basic and extended. The basic
+level comprises core APIs needed to perform SYCL offload. The extended level is
+needed only for OpenCL interoperability. 
+
+### The basic PI APIs
+
+This level requires availability of functions representing core functionality,
+such as device memory management, kernel creation and parameter setting,
+enqueuing kernel for execution, etc. There are functions which fully match
+semantics of the corresponding OpenCL X.X functions, as well as functions which
+don't have direct mapping to OpenCL. See the
+[pi.h](../include/CL/sycl/detail/pi.h) header for the full list and
+descriptions of PI APIs. [TBD: link to pi.h doxygen here]
+
+*__Examples of PI OpenCL-based APIs__*
+
+    piCreateKernel
+    piReleaseKernel
+    piSetKernelArg
+
+*__Example of PI non OpenCL-based APIs__*
+
+Selects the most appropriate device image based on runtime
+information and the image characteristics
+```
+cl_int piSelectDeviceImage(pi_context ctx, pi_device_image **images,
+                             cl_uint num_images,
+                             pi_device_image **selected_image);
+```
+
+PI also defines few types and string tags to describe a device binary image.
+Those are used to communicate to plugins information about the images where it
+is needed, currently only in the ```piSelectDeviceImage``` function. The main
+type is ```pi_device_image```, whose detailed description can also be found in
+the header.  The layout of this type strictly matches the layout of the corresponding
+device image descriptor type defined in the ```clang-offload-wrapper``` tool
+which wraps device binaries into a host object for further linkage. The wrapped
+images reside inside this descriptor in a data section.
+
+### The Extended PI APIs
+
+These are APIs needed to implement SYCL-to-OpenCL interoperability. Only
```

Comment:
I think this section should be generalized to describe extended PI APIs as a mechanism to provide interoperability with device specific API not **only OpenCL**.
SYCL should be able to interoperate with OpenMP if SYCL runtime uses extended OpenMP plug-in API. Right?

Or this section should be called "The extended OpenCL plugin API".

Reply:
> Or this section should be called "The extended OpenCL plugin API".

+1

Reply:
> with device specific API not only OpenCL.
this was specifically for OpenCL interop. Interop with any device is a big separate step